UnityProject/Assets/Scripts/HealthV2/Living/HealthStateController.cs

```diff
@@ -236,6 +256,61 @@ private void SyncHealthDoll(string oldDollData, string newDollData)
 			}
 		}
 
+		[TargetRpc]
+		private void InvokeClientHungerEvent(HungerState state)
+		{
+			HungerEvent?.Invoke(state);
+		}
+
+		[TargetRpc]
+		private void InvokeClientBleedEvent(BleedingState state)
+		{
+			BleedingEvent?.Invoke(state);
+		}
+
+		[TargetRpc]
+		private void InvokeClientFireStackEvent(float state)
+		{
+			FireStacksEvent?.Invoke(state);
+		}
+
+		[TargetRpc]
+		private void InvokeClientOverallHealthEvent(float state)
+		{
+			OverallHealthEvent?.Invoke(state);
+		}
+
+		[TargetRpc]
+		private void InvokeClientConsciousStateEvent(ConsciousState state)
+		{
+			ConsciousEvent?.Invoke(state);
+		}
+
+
+		[TargetRpc]
+		private void InvokeClientSufficationEvent(bool state)
+		{
+			SuffuicationEvent?.Invoke(state);
+		}
+
+		[TargetRpc]
+		private void InvokeClientToxinsEvent(bool state)
+		{
+			ToxinEvent?.Invoke(state);
+		}
+
+		[TargetRpc]
+		private void InvokeClientTemperatureEvent(float state)
+		{
+			TemperatureEvent?.Invoke(state);
+		}
+
+		[TargetRpc]
+		private void IvokeClientPressureEvent(float state)
+		{
+			PressureEvent?.Invoke(state);
+		}
```

Comment:
Just realised reading this through again that these would probably be better as SyncVar hooks.

If you have time to change them, otherwise i'm happy to merge it now and have it done in another PR?

Reply:
SyncVar hooks trigger on all clients if I remember correctly, TargetRpc only triggers on the owner.

Reply:
HealthStateController is special as theres a setting on the script on the prefab so that the hooks will only trigger for the connection that owns the object

Reply:
I can fix this in the next PR, right now I've opened a new branch and working on something else so i can move that there.